JS/Lesson32/onur/my-app/src/App.jsx

```diff
@@ -1,39 +1,30 @@
 import './App.css';
 import { useEffect, useState } from 'react';
-import {BD_SEARCH_BASE_URL} from './urls.js';
-import {RecipeList} from './components/RecipeList';
-import {SearchBar} from './components/SearchBar';
+import { BD_SEARCH_BASE_URL } from './urls.js';
+import { RecipeList } from './components/RecipeList';
+import { SearchBar } from './components/SearchBar';
+import { RecipeDetail } from './components/RecipeDetail/index.jsx';
+
 
 export const App = () => {
-  // 1. Fetch recipes data from an API (Get request, API key, useState to store data, useEffect)
-  // 2. Display the data (RecipeList component -> use <ul> to display data)
-  // 3. Search recipe (A new component SearchBar with a <form> element, text input, and submit functionality)
 
-	const [recipes, setRecipes] = useState([]);
-    // If you want to keep both initial recipes and recipes from search  on the screen at the same time use code below:
-  // const [searchResult, setSearchResult] = useState([]);
+  const [recipes, setRecipes] = useState([]);
 
-  // Runs once of component render (component render = "sayfanin yenilenmesi")
   useEffect(() => {
-    fetch(BD_SEARCH_BASE_URL)
-    .then(response => response.json())
-    .then(data => setRecipes(data.meals));
-  },[]);
 
-  // If you want to keep both initial recipes and recipes from search  on the screen at the same time use code below:
-  // useEffect(() => {
-  //   setRecipes(prevState => {
-  //     // compare searchResult with prevState of recipes
-  //     // only add values from searchResult array that are not present in the prevState array (recipes)
-  //   });
-
-  // },[recipes, searchResult]);
-  
-	return (
-		<div className='container'>
-      <SearchBar setRecipes={setRecipes}/>
-      <RecipeList recipes={recipes} />
-		</div>
-	);
-};
+    fetch(BD_SEARCH_BASE_URL)
+      .then(response => response.json())
+      .then(data => {
+        setRecipes(data.meals)
+        console.log(data)
+      })
+      .catch(error => console.error('Request canceled!'));
+  }, [RecipeList, SearchBar, RecipeDetail]);
```

Comment:
You don't need to add child components as dependencies to useEffect hook. There are a few reasons for it:
- Reactivity to Props and State: The re-rendering of child components is typically driven by changes in props or state. If your effect depends on these values, they should be the dependencies, not the child components themselves.
- Functions or Callbacks: If you pass functions (like event handlers or callbacks) to child components and these functions are defined in the parent component, it's the functions or the values they depend on that should be listed in the dependencies array, if they change over time.
- Stability of Components: Components themselves, as elements or types, are usually stable and don't change across renders. What changes are their props or the state that affects them.
- Complexity and Performance: Adding components to the dependencies array could lead to unnecessary complexity and might cause performance issues, as it's not straightforward how React would determine if a component in the dependency array has “changed”.

So here, we just need to fetch the data initially and save it. If there's an update from the SearchBar component, we'll re-fetch data there. If there's a change in the RecipeList or RecipeDetails components, it should not trigger a new fetch request at all.